### ISSUE TITLE
Too verbose deprecations warnings

### ISSUE THREAD

**Issue by purbon** [elastic/logstash#4444]
With this very simple config,

```
input {
  stdin {}
}

output {
  stdout { codec => rubydebug }
  elasticsearch {
    hosts => [ "127.0.0.1:9200" ]
  }
}
```

when I start logstash I see this warning message,

```
You are using a deprecated config setting "retry_max_items" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future.  If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"retry_max_items", :plugin=><LogStash::Outputs::ElasticSearch hosts=>["127.0.0.1:9200"], codec=><LogStash::Codecs::Plain charset=>"UTF-8">, workers=>1, index=>"logstash-%{+YYYY.MM.dd}", manage_template=>true, template_name=>"logstash", template_overwrite=>false, flush_size=>500, idle_flush_time=>1, doc_as_upsert=>false, max_retries=>3, script_type=>"inline", script_var_name=>"event", scripted_upsert=>false, retry_max_interval=>2, retry_max_items=>500, action=>"index", path=>"/", ssl=>false, ssl_certificate_verification=>true, sniffing=>false, sniffing_delay=>5>, :level=>:warn}
You are using a deprecated config setting "retry_max_items" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future.  If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"retry_max_items", :plugin=><LogStash::Outputs::ElasticSearch hosts=>["127.0.0.1:9200"], codec=><LogStash::Codecs::Plain charset=>"UTF-8">, workers=>1, index=>"logstash-%{+YYYY.MM.dd}", manage_template=>true, template_name=>"logstash", template_overwrite=>false, flush_size=>500, idle_flush_time=>1, doc_as_upsert=>false, max_retries=>3, script_type=>"inline", script_var_name=>"event", scripted_upsert=>false, retry_max_interval=>2, retry_max_items=>500, action=>"index", path=>"/", ssl=>false, ssl_certificate_verification=>true, sniffing=>false, sniffing_delay=>5>, :level=>:warn}
You are using a deprecated config setting "retry_max_items" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future.  If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"retry_max_items", :plugin=><LogStash::Outputs::ElasticSearch hosts=>["127.0.0.1:9200"], codec=><LogStash::Codecs::Plain charset=>"UTF-8">, workers=>1, index=>"logstash-%{+YYYY.MM.dd}", manage_template=>true, template_name=>"logstash", template_overwrite=>false, flush_size=>500, idle_flush_time=>1, doc_as_upsert=>false, max_retries=>3, script_type=>"inline", script_var_name=>"event", scripted_upsert=>false, retry_max_interval=>2, retry_max_items=>500, action=>"index", path=>"/", ssl=>false, ssl_certificate_verification=>true, sniffing=>false, sniffing_delay=>5>, :level=>:warn}
```

As I used the default values for ```retry_max_items```, I don't understand this warning message. I would understand this warning if I have used it explicitly. What do you think?

Another question is, why does it show up three times?

**Comment by guyboertje** [comment]
I also got many occurrences of:
```
You are using a deprecated config setting "name" set in redis
You are using a deprecated config setting "verify_ssl" set in http.
```
as well as the one the @purbon stated above.

**Comment by purbon** [comment]
this looks like something buggy going into the reporters

**Comment by guyboertje** [comment]
OK, I got this.

**Comment by guyboertje** [comment]
I found the culprit.  OutputDelegator.  PR soon.  Including a test is hard because the the output being wrapped is a test double and it will not exhibit the fault.

**Comment by guyboertje** [comment]
fixed with #4467

**Comment by guyboertje** [comment]
On reflection, the OutputDelegator is not really to blame.
